python-frontend/src/main/java/org/sonar/python/semantic/SymbolImpl.java

```diff
@@ -65,6 +68,17 @@ public SymbolImpl(String name, @Nullable String fullyQualifiedName, @Nullable St
     this.kind = Kind.OTHER;
   }
 
+  public SymbolImpl(SymbolsProtos.VarSymbol varSymbol) {
+    this.name = varSymbol.getName();
+    this.fullyQualifiedName = TypeShed.normalizedFqn(varSymbol.getFullyQualifiedName());
+    String fqn = varSymbol.getTypeAnnotation().getFullyQualifiedName();
+    if (!fqn.isEmpty()) {
```

Comment:
Related to the previous comment, what happens here on a serialized type name that is `#Unknown`?

Reply:
I think we can prevent this to happen in the serializer. We could add a check that would throw an exception if type name is `#Unknown`

Reply:
Thinking again about this: `#Unknown` is the `pretty_printed_name` which we currently don't really use in the deserializer. The fqn instead should always be empty.